plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlJdbcConfig.java

```diff
@@ -42,8 +41,11 @@ public boolean isUrlWithoutDatabase()
     {
         try {
             Driver driver = new Driver();
-            Properties properties = driver.parseURL(getConnectionUrl(), null);
-            return (properties == null) || (driver.database(properties) == null);
+            if (driver.acceptsURL(getConnectionUrl())) {
+                String path = ConnectionUrlParser.parseConnectionString(getConnectionUrl()).getPath();
+                return path == null || path.strip().equals("");
```

Comment:
Why do you call `strip` on this? Can it have empty strings? If not `nullOrEmpty(path)` might be more concise. Otherwise ignore.